### PR TITLE
Add field AreExternalFieldsFilled for AggregationFlowRecord

### PR DESCRIPTION
**Pull request by dreamtalen**
Some fields like Pod labels will be filled externally on the IPFIX Mediator, areExternalFieldsFilled is an indicator for IPFIX Mediator to check whether these fields has been filled. 
This field is set to false when creating new AggregationFlowRecord.
Setting and utilizing this field is upto the user and not used in go-ipfix library code.

### CODE REVIEW

File: pkg/intermediate/aggregate.go [hunk 1]
```diff
@@ -273,6 +273,14 @@ func (a *AggregationProcess) IsMetadataFilled(record AggregationFlowRecord) bool
 	return record.isMetaDataFilled
 }
 
+func (a *AggregationProcess) SetHasFilledOnFlowAggregator(record AggregationFlowRecord, isFilled bool) {
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:276]: Let's keep the name generic by calling `hasFilledExternalFields`.

**Review by srikartati — COMMENTED** [review]
Thanks for the change, Yongming.
It is not your change, but I feel `isMetaDataFilled` field name is not appropriate.
I think 'areCorrelatedFieldsFilled` is much more generic and apt for ipfix intermediate process.
What do you think @zyiou ?

**Comment by dreamtalen** [comment]
> Thanks for the change, Yongming.
> It is not your change, but I feel `isMetaDataFilled` field name is not appropriate.
> I think 'areCorrelatedFieldsFilled` is much more generic and apt for ipfix intermediate process.
> What do you think @zyiou ?

Thanks Srikar, checked with yiou offline and done the change.

File: pkg/intermediate/types.go [hunk 2]
```diff
@@ -33,11 +33,15 @@ type AggregationFlowRecord struct {
...
-	// isMetaDataFilled is an indicator for IPFIX Mediator to check whether K8s
+	// areCorrelatedFieldsFilled is an indicator for IPFIX Mediator to check whether K8s
 	// metadata are filled for flow record. It is always true for Intra-Node
 	// and ToExternal flows and only applicable for Inter-Node flows that are
 	// not required to be correlated. (e.g. flows with Egress deny rule applied)
-	isMetaDataFilled bool
+	areCorrelatedFieldsFilled bool
+	// Some fields like Pod labels will be filled on the Flow Aggregator side,
+	// hasFilledExternalFields is an indicator to check whether these fields
+	// has been filled on Flow Aggregator.
```
Comment (srikartati) [inline pkg/intermediate/types.go:43]: Could you rephrase this comment so that it is applicable for a generic IPFIX intermediate process and not specific to Flow Aggregator which is part of Antrea? 
Please mentioned that setting and utilizing this field is upto the user and not used in go-ipfix library code.

File: pkg/intermediate/types.go [hunk 3]
```diff
@@ -33,11 +33,15 @@ type AggregationFlowRecord struct {
...
+	hasFilledExternalFields bool
```
Comment (srikartati) [inline pkg/intermediate/types.go:44]: I am amending my suggestion. I realized that we can have both of them in the same format: `areExternalFieldsFilled`. Please consider this modification.

Comment (dreamtalen) [reply pkg/intermediate/types.go]: > I am amending my suggestion. I realized that we can have both of them in the same format: `areExternalFieldsFilled`. Please consider this modification.

Thanks, addressed. Please also take a look at https://github.com/vmware/go-ipfix/pull/212 Srikar, since we found limitation in current flowRecordMap data structure when adding these fields.

File: pkg/intermediate/types.go [hunk 4]
```diff
@@ -33,11 +33,17 @@ type AggregationFlowRecord struct {
...
 	ReadyToSend               bool
 	waitForReadyToSendRetries int
-	// isMetaDataFilled is an indicator for IPFIX Mediator to check whether K8s
+	// areCorrelatedFieldsFilled is an indicator for IPFIX Mediator to check whether K8s
 	// metadata are filled for flow record. It is always true for Intra-Node
 	// and ToExternal flows and only applicable for Inter-Node flows that are
 	// not required to be correlated. (e.g. flows with Egress deny rule applied)
-	isMetaDataFilled bool
+	areCorrelatedFieldsFilled bool
+	// Some fields like Pod labels will be filled externally on the IPFIX Mediator,
```
Comment (srikartati) [inline pkg/intermediate/types.go:41]: "Some fields could be filled externally on aggregation records once before exporting them in IPFIX mediator, e.g., metadata of a flow. areExternalFieldsFilled is an indicator for IPFIX Mediator to check whether
these fields has been filled or not before exporting. ....."
Prefer not to have Pod Labels.

**Review by srikartati — COMMENTED** [review]
One comment otherwise LGTM.

File: pkg/intermediate/types.go (hunk 4 above)
Comment (dreamtalen) [reply pkg/intermediate/types.go]: Thanks, addressed.

**Review by srikartati — DISMISSED** [review]
LGTM
I think you probably have to wait for Yiou's PR to merge.

**Comment by dreamtalen** [comment]
> LGTM
> I think you probably have to wait for Yiou's PR to merge.

Rebased after yiou's PR merged, could you please give me an approval again?

**Review by srikartati — APPROVED** [review]
